genfragments/ThirteenPointSixTeV/Higgs/HH/HHto2W2Zto4L_TuneCP5_13p6TeV_powheg-pythia8.py

```diff
@@ -0,0 +1,44 @@
+import FWCore.ParameterSet.Config as cms
+from Configuration.Generator.Pythia8CommonSettings_cfi import *
+from Configuration.Generator.MCTunesRun3ECM13p6TeV.PythiaCP5Settings_cfi import *
+from Configuration.Generator.PSweightsPythia.PythiaPSweightsSettings_cfi import *
+from Configuration.Generator.Pythia8PowhegEmissionVetoSettings_cfi import *
+
+generator = cms.EDFilter("Pythia8ConcurrentHadronizerFilter",
+                         maxEventsToPrint = cms.untracked.int32(1),
+                         pythiaPylistVerbosity = cms.untracked.int32(1),
+                         filterEfficiency = cms.untracked.double(1.0),
+                         pythiaHepMCVerbosity = cms.untracked.bool(False),
+                         comEnergy = cms.double(13600.),
+                         PythiaParameters = cms.PSet(
+        pythia8CommonSettingsBlock,
+        pythia8CP5SettingsBlock,
+        pythia8PSweightsSettingsBlock,
+        pythia8PowhegEmissionVetoSettingsBlock,
+        processParameters = cms.vstring(
+            'POWHEG:nFinal = 2',
+            '23:mMin = 0.05',
+            '23:onMode = off',
+            '23:onIfAny = 11 13 15', # only leptonic Z decays
+            '25:m0 = 125.0',
+            '25:onMode = off',
+            '25:onIfMatch = 24 24',
+            '25:onIfMatch = 23 23',
+            'ResonanceDecayFilter:filter = on',
+            'ResonanceDecayFilter:exclusive = off', #off: require at least the specified number of daughters, on: require exactly the specified number of daughters
+            'ResonanceDecayFilter:eMuAsEquivalent = off', #on: treat electrons and muons as equivalent
+            'ResonanceDecayFilter:eMuTauAsEquivalent = on', #on: treat electrons, muons , and taus as equivalent
+            'ResonanceDecayFilter:allNuAsEquivalent = off', #on: treat all three neutrino flavours as equivalent
+            'ResonanceDecayFilter:mothers = 23,24', # list of mothers not specified -> count all particles in hard process+resonance decays (better to avoid specifying mothers when including leptons from the lhe in counting, since intermediate resonances are not gauranteed to appear in general
+            'ResonanceDecayFilter:daughters = 11,11,11,11,11',
```

Comment:
> besides the comment inline, I want to ask those HH are all gluglu induced HH, right? why the dipolerecoil is needed since it's recommended for VBF like processes

Hi @menglu21 Thank you very much for your review! Those decay fragments with the dipolerecoil is for VBF HH production, so dipolerecoil is on following the recommendation for VBF like processes.

Reply:
The samples is H->ZZ->4l while the HWW is always semileptonic. So the "4L" in the name is related to 2Z. This decay fragment was adapted from the Run 2 one: https://cms-pdmv.cern.ch/mcm/public/restapi/requests/get_fragment/HIG-RunIISummer15wmLHEGS-02815

the Run 2 corresponding samples can be found on das: https://cmsweb.cern.ch/das/request?view=list&limit=50&instance=prod%2Fglobal&input=dataset%3D%2FGluGluToHHTo2W2ZTo4L_node_cHHH*13TeV-powheg-pythia8%2F*%2FNANOAODSIM